packages/gestalt/src/TextField/InternalTextField.js

```diff
@@ -25,6 +25,67 @@ import typography from '../Typography.css';
 
 type SizeType = 'sm' | 'md' | 'lg';
 
+type autoCompleteType =
```

Comment:
Can we use a shared definition between these since they are the same? Otherwise LGTM

Reply:
This is a known issue with the tool we use to autogenerate props tables from types — any types defined outside of a given file will show up as the type name rather than the type value. The gain of autogen types and a single source of truth for a component's types was worth the tradeoff of some type duplication.

That said, that would be an awesome problem to solve someday.

Reply:
Oh, but actually we can dedupe this type. The type needs to be defined in TextField.js to be picked up by the autogen tool. That type can be exported and consumed by InternalTextField.js. (Philosophically that's kinda weird/gross, but it might be preferable to repeating the type and the eventual drift that could happen)

Reply:
This was done this way because our docs are autogenerated from the component's props, and this autogeneration breaks when importing in the types. However since in the new approach we are overriding this autogeneration, I used the shared definition within InternalTextField :)